### PR TITLE
fix HttpClient severe performance degradation by use threadPool in se…

### PR DESCRIPTION
**Pull request by novain1102**
…ndImmdiate

Re: #https://github.com/cocos/3d-tasks/issues/13258

### Changelog

*fix HttpClient severe performance degradation by use threadPool in sendImmdiate

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by fish9167** [comment]
@cocos-robot run test cases

File: native/cocos/network/HttpClient-apple.mm [hunk 1]
```diff
@@ -345,6 +346,7 @@ static int processTask(HttpClient *client, HttpRequest *request, NSString *reque
 
 HttpClient::~HttpClient() {
     CC_SAFE_RELEASE(_requestSentinel);
+    CC_SAFE_DELETE(_threadPool);
```
Comment (dumganhar) [inline native/cocos/network/HttpClient-apple.mm]: Deleting thread pool will join all threads. If network connection is bad, join may cost a lot of time. I think keeping it as a global variable is ok.

Comment (novain1102) [reply native/cocos/network/HttpClient-apple.mm]: make sense. Done!

File: native/cocos/network/HttpClient.h [hunk 2]
```diff
@@ -31,6 +31,7 @@
 #include <condition_variable>
 #include <thread>
 #include "base/RefVector.h"
+#include "base/ThreadPool.h"
```
Comment (dumganhar) [inline native/cocos/network/HttpClient.h]: Remove this line , include it in cpp file.

Comment (novain1102) [reply native/cocos/network/HttpClient.h]: done

File: native/cocos/network/HttpClient-apple.mm (hunk 1 above)
Comment (minggo) [reply native/cocos/network/HttpClient-apple.mm]: Should refactor it after network module is added. Threads are a limitation resource.

File: native/cocos/network/HttpClient-java.cpp [hunk 3]
```diff
@@ -683,7 +685,7 @@ void HttpClient::processResponse(HttpResponse *response, char *responseMessage)
     urlConnection.disconnect();
 
     // write data to HttpResponse
-    response->setResponseCode(static_cast<long>(responseCode));
+    response->setResponseCode(static_cast<int32_t>(responseCode));
```
Comment (dumganhar) [inline native/cocos/network/HttpClient-java.cpp]: Why cast it to int32_t ?

```c++
    inline void setResponseCode(long value) {
        _responseCode = value;
    }
```

**Review by minggo — APPROVED** [review]

File: native/cocos/network/HttpClient-java.cpp (hunk 3 above)
Comment (novain1102) [reply native/cocos/network/HttpClient-java.cpp]: change to //NOLINT to passby `long` type clang-tidy

**Review by dumganhar — APPROVED** [review]

**Review by PatriceJiang — APPROVED** [review]

**Pull request merged**